src/components/profile/Profile.js

```diff
@@ -48,6 +68,29 @@ const Profile = props => {
     }
   }, [token, loggedInUser, dispatch]);
 
+  const leaveAllegiance = async id => {
```

Comment:
i think this is a good reason to change the backend for user/allegiance, group/user, group/allegiance relations. the delete function should be able to take in the user id/allegiance id instead of only the relation. seems silly to require 2 network calls to leave an allegiance.

Reply:
Great point, consider it done.